Compilation failures with clang on Fedora 37

There is multiple issues:

* First, multiple instances of https://stackoverflow.com/questions/33855949/why-is-the-compiler-showing-call-to-function-with-no-prototype. Saying `void foo(void)` appears to be the correct standard-compliant C, https://stackoverflow.com/questions/693788/is-it-better-to-use-c-void-arguments-void-foovoid-or-not-void-foo.

```
/usr/bin/clang  -I/home/jdanek/repos/skupper-router/include -I/home/jdanek/repos/skupper-router/cmake-build-relwithdebinfo-clang/include -I/usr/include/python3.11 -I/home/jdanek/repos/skupper-router/cmake-build-relwithdebinfo-clang/src -I/home/jdanek/repos/skupper-router/src -I/home/jdanek/repos/skupper-router/src/router_core -isystem /home/jdanek/repos/qpid/qpid-proton/build/install/include -O2 -g -DNDEBUG -fcolor-diagnostics -Wall -Wextra -Werror -Wpedantic -Wno-unused-parameter -Wno-missing-field-initializers -Wno-sign-compare -Wno-language-extension-token -gdwarf-aranges -std=gnu11 -MD -MT src/CMakeFiles/skupper-router.dir/adaptors/adaptor_common.c.o -MF src/CMakeFiles/skupper-router.dir/adaptors/adaptor_common.c.o.d -o src/CMakeFiles/skupper-router.dir/adaptors/adaptor_common.c.o -c /home/jdanek/repos/skupper-router/src/adaptors/adaptor_common.c
In file included from /home/jdanek/repos/skupper-router/src/adaptors/adaptor_common.c:19:
In file included from /home/jdanek/repos/skupper-router/src/adaptors/adaptor_common.h:24:
In file included from /home/jdanek/repos/skupper-router/src/entity.h:22:
/home/jdanek/repos/skupper-router/include/qpid/dispatch/error.h:83:39: error: a function declaration without a prototype is deprecated in all versions of C [-Werror,-Wstrict-prototypes]
QD_EXPORT const char* qd_error_message();
                                      ^
                                       void
```

* Second, some options for tests subfolders that are unknown in clang

```
FAILED: tests/c_unittests/CMakeFiles/c_unittests.dir/test_terminus.cpp.o 
/usr/bin/clang++ -Dc_unittests_EXPORTS -I/home/jdanek/repos/skupper-router/include -I/home/jdanek/repos/skupper-router/cmake-build-relwithdebinfo-clang/include -I/home/jdanek/repos/qpid/qpid-proton/build/install/include -I/usr/include/python3.11 -I/home/jdanek/repos/skupper-router/src -I/home/jdanek/repos/skupper-router/cmake-build-relwithdebinfo-clang/src -I/home/jdanek/repos/skupper-router/src/router_core -I/home/jdanek/repos/skupper-router/tests/cpp-stub -fno-inline -fno-builtin -fno-stack-protector -Wno-literal-suffix -O2 -g -DNDEBUG -fcolor-diagnostics -Wall -Wextra -Werror -Wpedantic -Wno-unused-parameter -Wno-missing-field-initializers -Wno-sign-compare -Wno-language-extension-token -gdwarf-aranges -std=c++17 -MD -MT tests/c_unittests/CMakeFiles/c_unittests.dir/test_terminus.cpp.o -MF tests/c_unittests/CMakeFiles/c_unittests.dir/test_terminus.cpp.o.d -o tests/c_unittests/CMakeFiles/c_unittests.dir/test_terminus.cpp.o -c /home/jdanek/repos/skupper-router/tests/c_unittests/test_terminus.cpp
error: unknown warning option '-Wno-literal-suffix'; did you mean '-Wno-literal-range'? [-Werror,-Wunknown-warning-option]
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -35,8 +35,12 @@ set(CMAKE_ENABLE_EXPORTS TRUE)
 
 
 # Set warning compile flags
-set(C_WARNING_GNU -Wall -Wextra -Werror -Wpedantic -pedantic-errors -Wimplicit-fallthrough=3 -Wno-empty-body -Wno-unused-parameter  -Wno-missing-field-initializers -Wno-sign-compare -Wno-type-limits)
-set(C_WARNING_Clang -Wall -Wextra -Werror -Wpedantic -Wno-unused-parameter -Wno-missing-field-initializers -Wno-sign-compare -Wno-language-extension-token)
+set(C_WARNING_GNU -Wall -Wextra -Werror -Wpedantic -pedantic-errors
+        $<$<COMPILE_LANGUAGE:C>:-Wstrict-prototypes> $<$<COMPILE_LANGUAGE:C>:-Wold-style-definition> -Wimplicit-fallthrough=3
+        -Wno-empty-body -Wno-unused-parameter  -Wno-missing-field-initializers -Wno-sign-compare -Wno-type-limits)
+set(C_WARNING_Clang -Wall -Wextra -Werror -Wpedantic
+        -Wstrict-prototypes -Wold-style-definition
+        -Wno-unused-parameter -Wno-missing-field-initializers -Wno-sign-compare -Wno-language-extension-token)
 
 # Additional compiler-specific flags
 #  -gdwarf-aranges to workaround https://sourceware.org/bugzilla/show_bug.cgi?id=22288 (backtrace line printing with libdw and clang)

--- a/src/posix/symbolization_none.c
+++ b/src/posix/symbolization_none.c
@@ -31,4 +31,4 @@ void qd_print_symbolized_backtrace_line(FILE *dump_file, const char *fallback_sy
     fprintf(dump_file, "   %s\n", fallback_symbolization);
 }
 
-void qd_symbolize_finalize() {}
+void qd_symbolize_finalize(void) {}

--- a/tests/adaptor_buffer_test.c
+++ b/tests/adaptor_buffer_test.c
@@ -23,7 +23,7 @@
 
 void qd_log_initialize(void);
 void qd_log_finalize(void);
-void qd_error_initialize();
+void qd_error_initialize(void);
 void qd_router_id_finalize(void);
 
 static char *test_qd_adaptor_buffer(void *context)
@@ -330,7 +330,7 @@ static char *test_qd_adaptor_copy_qd_buffers_to_adaptor_buffers(void *context)
     return 0;
 }
 
-int adaptor_buffer_tests()
+int adaptor_buffer_tests(void)
 {
     int   result     = 0;
     char *test_group = "adaptor_buffer_tests";

